swap/swap_test.go

```diff
@@ -519,24 +550,19 @@ func newTestSwap(t *testing.T, key *ecdsa.PrivateKey) (*Swap, func()) {
 	return swap, clean
 }
 
-type dummyPeer struct {
-	*protocols.Peer
-}
-
-// creates a dummy protocols.Peer with dummy MsgReadWriter
-func newDummyPeer() *dummyPeer {
-	return newDummyPeerWithSpec(nil)
+// creates a dummy swap.Peer
+func newDummyPeer() *Peer {
```

Comment:
Are you making these changes? It doesn't sound reading the PR description, are these changes from some other PR? In this case we need to look at how you rebased.

The PR should only contain your changes.
If you did these changes, then disregard.

Reply:
I made these changes. The main reason is that I was looking at it and I had the idea it was making things overly complex. 

If you feel these changes should be removed, as it was not part of the original PR, I don't have hard feelings about reverting them.

Reply:
Do we even need the dummyPeer structure at all at this point? It seems the only thing were doing with it is accessing `.Peer` and passing that to other functions. In that case why not just return the protoPeer directly?

Reply:
So first of all the suggestion is to not add things to your PR which are not part of your work package. This helps keeping a PR small! Unless of course it is somehow related or a very small fix (e.g. oneliner).

Now with @ralph-pichler 's comment even more, it makes clear we are dragging other issues into this PR. Let's remove this change from this PR and address it, if needed, in a different one. 

It was called `...WithSpec` because it is not just a `protoPeer`, but a specific one as we need the `Spec` from the `swap` protocol associated with it, which we don't get by default. Now it may be that we don't even need it anymore, I'd have to check, that's why I'd prefer we keep this change out of this PR.

Reply:
@holisticode , I agree with you that it would be a good idea to assess whether the dummyPeer structure is needed at all. 

It took me personally some time to figure out what this dummyPeer is and how to work with it after I figured out, I simplified the structure, just to make it more understandable for me. Now it indeed it looks like it can even be simpler.

For the sake of our future self and future collaborators, I  think we should keep it out for now, and only add the feature when we are actually going to use it.

Yet, for now, I will remove it as indeed it is not related to this PR and I don't want to drag it along.